### DISCUSSION TITLE
(nxViteTsPaths) nx-tsconfig-paths.plugin should support different tsconfig.json per project

### DISCUSSION THREAD

**Discussion by hivaga** [nrwl/nx#23294] [Feature Requests]
https://github.com/nrwl/nx/blob/master/packages/vite/plugins/nx-tsconfig-paths.plugin.ts

It looks this plugin (nxViteTsPaths)  does not support/ignores any customisation for the project's build tsconfig.json. The code as it is here is hardcoded to always fallback/use the [monorepo-root]/tsconfig.base.json. This is not  ideal and it is also super confusing for the developers because in many places like, project.json, vite.config.ts there is an option to specify the use tsconfig.json but aperantly this is ignored and the plugin does not allow this sort of customisation.

This leads to a problem where you try to configure the build in a certain way but you are constrain to have single tsconfig.json for your build which in fact is the [monorepo-root]/tsconfig.base.json

**Comment by indieisaconcept** [comment]
Just discovered this myself when using `vitest` as tests were failing to resolve aliased imports. The fix seems simple enough. `getTsConfig` just needs to have the projectRoot passed as an additional argument when called and an extra resolve added which makes use of it.

Happy to submit a PR

**Before**
https://github.com/nrwl/nx/blob/a3869a814dcf0f4c3d6fc2d2662e970273ccdb03/packages/vite/plugins/nx-tsconfig-paths.plugin.ts#L200-L211

**After**
```javascript
 function getTsConfig(preferredTsConfigPath: string, projectRoot: string): string { 
   return [ 
     resolve(preferredTsConfigPath), 
     resolve(join(projectRoot, 'tsconfig.json')), 
     resolve(join(workspaceRoot, 'tsconfig.base.json')), 
     resolve(join(workspaceRoot, 'tsconfig.json')), 
   ].find((tsPath) => { 
     if (existsSync(tsPath)) { 
       logIt('Found tsconfig at', tsPath); 
       return tsPath; 
     } 
   }); 
 } 
 ```
